src/Microsoft.Kusto.ServiceLayer/DataSource/DstsAuthenticationManager.cs

```diff
@@ -0,0 +1,72 @@
+using System;
+using System.Collections.Concurrent;
+using System.Diagnostics;
+using System.IO;
+
+namespace Microsoft.Kusto.ServiceLayer.DataSource
+{
+    public class DstsAuthenticationManager
```

Comment:
This is where all the magic happens. Loads the dSTS Token from the ConsoleApp and saves it in a ConcurrentDictionary. The token is stored by `ClusterName Databasename -> ExpirationDate, Token`. We're unable to get the actual token expiration because it's not exposed.